homeassistant/components/recorder/services.py

```diff
@@ -135,10 +157,79 @@ async def async_handle_disable_service(service: ServiceCall) -> None:
     )
 
 
+@callback
+def _async_register_get_statistics_service(
+    hass: HomeAssistant, instance: Recorder
+) -> None:
+    async def async_handle_get_statistics_service(
+        service: ServiceCall,
+    ) -> ServiceResponse:
+        """Handle calls to the get_statistics service."""
+        start_time = dt_util.as_utc(service.data["start_time"])
+        end_time = (
+            dt_util.as_utc(service.data["end_time"])
+            if "end_time" in service.data
+            else None
+        )
+
+        statistic_ids = service.data.get("statistic_ids")
+        statistic_ids_set = set(statistic_ids) if statistic_ids else None
+
+        period = service.data["period"]
+        units = service.data.get("units")
+
+        if (types := service.data.get("types")) is None:
+            types = {"change", "last_reset", "max", "mean", "min", "state", "sum"}
+        else:
+            types = set(types)
+
+        result = await instance.async_add_executor_job(
+            statistics_during_period,
+            hass,
+            start_time,
+            end_time,
+            statistic_ids_set,
+            period,
+            units,
+            types,
+        )
+
+        formatted_result = {}
+        for statistic_id, statistic_rows in result.items():
+            formatted_statistic_rows = []
+
+            for row in statistic_rows:
+                formatted_row = {
+                    **row,
+                    "start": dt_util.utc_from_timestamp(row["start"]).isoformat(),
+                    "end": dt_util.utc_from_timestamp(row["end"]).isoformat(),
+                }
+
+                if last_reset := row.get("last_reset") is not None:
+                    formatted_row["last_reset"] = dt_util.utc_from_timestamp(
+                        last_reset
+                    ).isoformat()
+
+                formatted_statistic_rows.append(formatted_row)
+
+            formatted_result[statistic_id] = formatted_statistic_rows
+
+        return cast(ServiceResponse, formatted_result)
```

Comment:
This cast() feels wrong, but I don't understand how else I can get mypy to understand that this is fine.
Or maybe it isn't and I'm missing something, but it does work when testing.

Reply:
Should it even convert the timestamps to strings or would it make more sense to provide the data as-is?

As strings, it is consistent with what I see when I use the calendar.get_events service, but is that how it should be?